### ISSUE TITLE
does the trainer's learning rate drop automatically?

### ISSUE THREAD

**Issue by yanchao12122** [davisking/dlib#894]
I train face recog using 700K samples(6K IDS) over 100k steps, 
but the learning rate is still 0.1

// Here we do the training.  We keep passing mini-batches to the trainer until the
    // learning rate has dropped low enough.
    while(trainer.get_learning_rate() >= 1e-4)
    {
        qimages.dequeue(images);
        qlabels.dequeue(labels);
        trainer.train_one_step(images, labels);
    }

**Comment by yanchao12122** [comment]
ah, I misunderstand the parameters of 'steps without apparent progress',
I have seen the rate drops already
